mozci/push.py

```diff
@@ -32,7 +32,7 @@
 
 BASE_INDEX = "gecko.v2.{branch}.revision.{rev}"
 
-MAX_DEPTH = 14
+MAX_DEPTH = 31
```

Comment:
Isn't this a bit much? Shouldn't we just do 20 for the backstop?

Reply:
As discussed on Matrix, there are still cases (e.g. DONTBUILD) where we could go more than 20 pushes without a full back-stop, so 31 is to be safe (though it could even be larger than that in practice, it should be pretty rare). Moreover, it is also useful to find pre-existing intermittents in parents.

We don't want things to be too slow though, so we decided to make this configurable, and, as a follow-up, to add an option to use the previous backstop too (checking parameters.yml from the decision tasks.